lib/components/MonthPicker/MonthPicker.tsx

```diff
@@ -0,0 +1,252 @@
+  onFocus?: (event: FocusEvent) => void;
+  minYear?: number;
+  maxYear?: number;
+  ascendingYears?: boolean;
+}
+
+const months = [
+  { value: '1', label: 'Jan' },
+  { value: '2', label: 'Feb' },
+  { value: '3', label: 'Mar' },
+  { value: '4', label: 'Apr' },
+  { value: '5', label: 'May' },
+  { value: '6', label: 'Jun' },
+  { value: '7', label: 'Jul' },
+  { value: '8', label: 'Aug' },
+  { value: '9', label: 'Sep' },
+  { value: '10', label: 'Oct' },
+  { value: '11', label: 'Nov' },
+  { value: '12', label: 'Dec' },
+];
+const getMonths = () =>
+  months.map(month => (
+    <option value={month.value} key={month.value}>
+      {month.label}
+    </option>
+  ));
+
+const getYears = (min: number, max: number, ascending: boolean) => {
+  const start = ascending ? min : max;
+  const end = ascending ? max + 1 : min - 1;
+
+  return range(start, end).map(year => {
+    const yearStr = String(year);
+
+    return (
+      <option value={yearStr} key={yearStr}>
+        {yearStr}
+      </option>
+    );
+  });
+};
+
+const currYear = new Date().getFullYear();
+const renderNativeInput = isMobile({ tablet: true });
+const monthRef = createRef<HTMLSelectElement>();
+const yearRef = createRef<HTMLSelectElement>();
+
+const customValueToString = ({ month, year }: MonthPickerValue) =>
+  month && year ? `${year}-${month < 10 ? '0' : ''}${month}` : undefined;
+
+const stringToCustomValue = (value: string) => {
+  const [year, month] = value.split('-');
+
+  return {
+    month: parseInt(month, 10),
+    year: parseInt(year, 10),
+  };
+};
+
+type FieldType = keyof MonthPickerValue | 'native';
+const nativeToCustomValue = (
+  nativeValue: string,
+  value: MonthPickerValue,
+  fieldType: FieldType,
+) =>
+  ({
+    month: {
+      year: value && value.year ? value.year : undefined,
+      month: parseInt(nativeValue, 10) || undefined,
+    },
+    year: {
+      month: value && value.month ? value.month : undefined,
+      year: parseInt(nativeValue, 10) || undefined,
+    },
+    native: stringToCustomValue(nativeValue),
+  }[fieldType]);
+
+const handleChange = <Element extends HTMLSelectElement | HTMLInputElement>(
+  { onChange, value }: Pick<MonthPickerProps, 'onChange' | 'value'>,
+  fieldType: FieldType,
+) => (event: ChangeEvent<Element>) => {
+  if (typeof onChange === 'function') {
+    onChange(nativeToCustomValue(event.target.value, value, fieldType), event);
+  }
+};
+
+const handleBlur = <Element extends HTMLSelectElement | HTMLInputElement>(
+  { onBlur, value }: Pick<MonthPickerProps, 'onBlur' | 'value'>,
+  fieldType: FieldType,
+) => (event: FocusEvent<Element>) => {
+  if (typeof onBlur === 'function') {
+    const isRelatedTargetExternal =
+      event.relatedTarget !== monthRef.current &&
+      event.relatedTarget !== yearRef.current;
+
+    if (event.relatedTarget === null || isRelatedTargetExternal) {
+      onBlur(nativeToCustomValue(event.target.value, value, fieldType), event);
```

Comment:
Why do we pass the value to the blur handler? Is it deliberate that we're also *not* passing it to the focus handler?

Reply:
As discussed, it was for backwards compatibility with previous style guides, but we have decided to not pass any args to the `blur` and `focus` handlers, while also only passing the custom value to the `change` handler.